clicache/src/Serializable.cpp

```diff
@@ -544,7 +544,7 @@ namespace Apache
 
         _GF_MG_EXCEPTION_TRY2
           CacheImpl *cacheImpl = CacheRegionHelper::getCacheImpl(cache->GetNative().get());
-          cacheImpl->getSerializationRegistry()->addType((native::Serializable*(*)())System::Runtime::InteropServices::Marshal::GetFunctionPointerForDelegate(nativeDelegate).ToPointer());
+          cacheImpl->getSerializationRegistry()->addType((std::shared_ptr<native::Serializable>(*)())System::Runtime::InteropServices::Marshal::GetFunctionPointerForDelegate(nativeDelegate).ToPointer());
```

Comment:
Could we do a c++ style cast to `TypeFactoryMethod`?  There's a couple more you touched in this file where we might be able to make the change.

Reply:
I don't think so, TypeFactoryMethod is defined in terms of std::function but maybe.